src/app/containers/app-header/app-header.component.spec.ts

```diff
@@ -266,8 +270,84 @@ describe('AppHeaderComponent', () => {
 
   describe('getApplicationThemeForUser()', () => {
     it('get default theme when no roles', () => {
-      const themes = component.getApplicationThemeForUser([]);
+      const themes = component.getApplicationThemeForUser([], []);
       expect(themes).toEqual(AppConstants.DEFAULT_USER_THEME);
     });
+
+    it('should correct run error handling of logErrorAndReturnDefaultTheme()', () => {
+      const exampleError = new Error('400');
+      expect(component.logErrorAndReturnDefaultTheme(exampleError)).toBe(component.defaultTheme);
+      expect(loggerServiceMock.error).toHaveBeenCalledWith(exampleError);
+    });
+  });
+
+  describe('observable testing', async () => {
+    const userInfo: UserInfo = { id: '1', forename: 'Test', surname: 'User', email: 'testemail', active: true, roles: ['pui-case-manager'] };
+    const userDetails: UserDetails = {
+      sessionTimeout: {
+        idleModalDisplayTime: 100,
+        totalIdleTime: 0,
+      },
+      canShareCases: true,
+      userInfo
+    };
+    const applicationThemes = AppConstants.APPLICATION_USER_THEMES;
+    it('should allow running of combinelatest', async () => {
+      // set boolean to check combinelatest runs on the two observables correctly
+      let checkCombineLatestRuns: boolean = false;
```

Comment:
Is checking combineLatestRuns necessary?

Reply:
Yes please. We are waiting for 2 observables here [themes and userRoles]. We need to know its working only after 2 observables